src/EnergyPlus/DataWater.cc

```diff
@@ -122,6 +122,11 @@ namespace DataWater {
     bool WaterSystemGetInputCalled(false); // set true once input data gotten.
     bool AnyIrrigationInModel(false);      // control flag set true if irrigation input for ecoroof DJS PSU Dec 2006
 
+    void clear_state()
+    {
+        WaterSystemGetInputCalled = false;
```

Comment:
This is a good start, but the clear_state should reset all of the non-const variables above.